src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateGeneratorTestBase.cs

```diff
@@ -0,0 +1,233 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.Extensions.DependencyModel.Resolution;
+
+namespace Microsoft.AspNetCore.Http.SourceGeneration.Tests;
+
+public class RequestDelegateGeneratorTestBase
```

Comment:
You can use the testing library.

Reply:
I looked into the testing library and it seems like it would be the most helpful for "snapshot" style tests where we want to assert on the generated output.

The goal with this test scaffold is to:

- Provide infrastructure for validating the intermediary model produced by the generator
- Provide infrastructure for running and validating the behavior of the generated code

Do you think it's feasible to achieve the same with the built-in library without as much of the same overhead?

Reply:
Not sure if it's possible. Probably fine to keep this for now and change later if necessary.